### PR TITLE
Implement IDisposable on Browser and Page

### LINKED ISSUE #85: Implement IDisposable on Browser and Page

### PR DESCRIPTION
**Pull request by kblok**
closes #85

### CODE REVIEW

**Comment by kblok** [comment]
Ready to review @Meir017.

We this guy and #83 we release v0.3

File: lib/PuppeteerSharp/Page.cs [hunk 1]
```diff
@@ -446,10 +446,13 @@ private Page(Session client, Target target, FrameTree frameTree, bool ignoreHTTP
 
         public async Task CloseAsync()
         {
-            await _client?.Connection?.SendAsync("Target.closeTarget", new
+            if (_client != null && !_client.Connection.IsClosed)
```
Comment (Meir017) [inline lib/PuppeteerSharp/Page.cs]: could `_client.Connection` be `null`?

Comment (kblok) [reply lib/PuppeteerSharp/Page.cs]: It shouldn't be we could double check.

Comment (kblok) [reply lib/PuppeteerSharp/Page.cs]: Would this be clear to you?
`if (!(_client?.Connection?.IsClosed ?? true))`

Comment (Meir017) [reply lib/PuppeteerSharp/Page.cs]: I was thinking of suggesting something like that

Comment (kblok) [reply lib/PuppeteerSharp/Page.cs]: Awesome!

Comment (kblok) [reply lib/PuppeteerSharp/Page.cs]: How about now?

**Pull request merged**